File: go/words/tests/numerical_test.go
package tests

import (
	"testing"

	. "example.com/words/numerical"
)

func TestFibonacci(t *testing.T) {
	cases := []struct {
		n        int
		expected int
	}{
		{-3, -1},
		{10, 55},
		{15, 610},
	}
	for _, c := range cases {
		result := Fibonacci(c.n)
		if result != c.expected {
			t.Errorf("Fibonacci(%d) = %d; want %d", c.n, result, c.expected)
		}
	}
}

func TestFactorial(t *testing.T) {
	cases := []struct {
		n        int
		expected int
	}{
		{-3, -1},
		{5, 120},
		{10, 3628800},
	}
	for _, c := range cases {
		result := Factorial(c.n)
		if result != c.expected {
			t.Errorf("Factorial(%d) = %d; want %d", c.n, result, c.expected)
		}
	}
}

func TestGCD(t *testing.T) {
	cases := []struct {
		a        int
		b        int
		expected int
	}{
		{48, 18, 6},
		{100, 25, 25},
		{17, 13, 1},
	}
	for _, c := range cases {
		result := GCD(c.a, c.b)
		if result != c.expected {
			t.Errorf("GCD(%d, %d) = %d; want %d", c.a, c.b, result, c.expected)
		}
	}
}

func TestLCM(t *testing.T) {
	cases := []struct {
		a        int
		b        int
		expected int
	}{
		{4, 6, 12},
		{21, 6, 42},
		{7, 3, 21},
	}
	for _, c := range cases {
		result := LCM(c.a, c.b)
		if result != c.expected {
			t.Errorf("LCM(%d, %d) = %d; want %d", c.a, c.b, result, c.expected)
		}
	}
}

File: rust/words/tests/numerical_test.rs
use words::numerical::*;

#[cfg(test)]
mod tests {
    use super::*;

    #[test]
    fn test_fibonacci() {
        assert_eq!(fibonacci(-3), -1);
        assert_eq!(fibonacci(10), 55);
        assert_eq!(fibonacci(15), 610);
    }

    #[test]
    fn test_factorial() {
        assert_eq!(factorial(-3), -1);
        assert_eq!(factorial(5), 120);
        assert_eq!(factorial(10), 3628800);
    }

    #[test]
    fn test_gcd() {
        assert_eq!(gcd(48, 18), 6);
        assert_eq!(gcd(100, 25), 25);
        assert_eq!(gcd(17, 13), 1);
    }

    #[test]
    fn test_lcm() {
        assert_eq!(lcm(6, 4), 12);
        assert_eq!(lcm(21, 6), 42);
        assert_eq!(lcm(7, 3), 21);
    }
}